mypy/stubgenc.py

```diff
@@ -201,7 +201,16 @@ def strip_or_import(typ: str, module: ModuleType, imports: List[str]) -> str:
         imports: list of import statements (may be modified during the call)
     """
     stripped_type = typ
-    if module and typ.startswith(module.__name__ + '.'):
+    if any(c in typ for c in '[,'):
+        for subtyp in re.split(r'\[|,\]', typ):
+            strip_or_import(subtyp.strip(), module, imports)
+        if module:
+            stripped_type = re.sub(
+                r'(^|[\[, ]+)' + re.escape(module.__name__ + '.'),
+                r'\1',
+                typ,
+            )
```

Comment:
It would be nice to have some unit tests for the new functionality, since this logic is non-trivial. You can add them to `mypy/test/teststubgen.py`.

(Maybe also modify this so that the module name is passed instead of a module object.)

Reply:
Ah I couldn't find the tests for stubgen. I'll get those added. Thanks for the review!